frontend/src/components/ChristmasCanvas.svelte

```diff
@@ -8,13 +8,13 @@
   const [canvasx, canvasy] = [960, 960];
   let canvas;
 
-  type Pixel = { id: string; x: number; y: number; color: string };
+  type Pixel = { canvas: number; x: number; y: number; color: number };
 
   const socket = Socket((import.meta as any).env.VITE_BACKEND_URL);
 
-  const drawPixel = (x: number, y: number, color: string) => {
+  const drawPixel = (x: number, y: number, color: number) => {
     const ctx = canvas.getContext("2d");
-    ctx.fillStyle = color;
+    ctx.fillStyle = `#${color}`;
```

Comment:
<img width="1265" alt="image" src="https://github.com/flatypus/ChristmasTreeDecoration/assets/68029599/16a2ea0c-bf54-4d8a-912b-d1eebae9a421">
When I first load up the map, it looks like this (assumed because of ur parseInt(...16), but not loaded back with color.toString(16). But even when I do that, it has some odd colors:

<img width="1265" alt="image" src="https://github.com/flatypus/ChristmasTreeDecoration/assets/68029599/f299eded-492f-4304-aac4-6241f5536909">

And when placing colors, it acts oddly...

https://github.com/flatypus/ChristmasTreeDecoration/assets/68029599/b56682a4-f362-43f2-a8cc-3e9eb10b84dc

Reply:
I'm able to repro. I thought I fixed this in this branch but apparently it was a different one. The colours look correct to me but there's weirdness with which pixels are loaded - in the 2022 map, f.e., the top-left QR code is gone, as is the table in the bottom-right. I suspect this might be a bug in the code I used to migrate the db - I had similar issues while I was doing it.

Reply:
Fixed! Forgot to `padStart()` so something like `#000000` was becoming `#0` on decode

Reply:
ah ya, i was wondering if it was cuz of the 00s in color! makes sense